dragonfly/implementations/roach_daq_run_interface.py

```diff
@@ -264,29 +260,151 @@ def central_frequency(self):
-        result = self.provider.cmd(self.daq_target, 'set_central_frequency', payload=payload)
+        payload = {'cf':float(cf), 'channel':self.channel_id}
+        result = self.provider.cmd(self.daq_target, 'set_central_frequency', payload = payload)
         logger.info('The roach central frequency is now {}Hz'.format(result['values'][0]))
+
         # The roach frequency can differ from the requested frequency. Psyllid should have the same frequency settings as the ROACH
         self.freq_dict[self.channel_id]=round(result['values'][0])
-        payload={'channel':self.channel_id, 'cf':self.freq_dict[self.channel_id], 'channel':self.channel_id}
-        result = self.provider.cmd(self.psyllid_interface, 'set_central_frequency', payload=payload)
+        payload = {'channel':self.channel_id, 'cf':self.freq_dict[self.channel_id], 'channel':self.channel_id}
+        result = self.provider.cmd(self.psyllid_interface, 'set_central_frequency', payload = payload)
         if result['values'][0]!=True:
             logger.warning('Could not set central frequency in Psyllid')
             self.freq_dict[self.channel_id]=None
             raise core.exceptions.DriplineGenericDAQError('Could not set central frequency in Psyllid stream')
 
-    # This method should not be used yet
-    def make_trigger_mask(self, snr, filename = '/home/roach/trigger_masks/psyllid_trigger_mask'):
-        # Set threshold snr
-        payload = {'channel':self.channel_id, 'snr':snr}
-        self.provider.cmd(self.psyllid_interface, 'set_fmt_snr_threshold', payload = payload)
-        time.sleep(1)
-        # Acquire and save new mask
-        timestr = time.strftime("%Y%m%d-%H%M%S")
-        filename = '{}_channel_{}_{}.json'.format(filename, self.channel_id, timestr)
-        payload = {'channel':self.channel_id, 'filename':filename}
-        result = self.provider.cmd(self.psyllid_interface, 'make_trigger_mask', payload=payload)
-        if result['values'][0]!=True:
+
+    # trigger control
+    @property
+    def trigger_type(self):
+        result = self.provider.cmd(self.psyllid_interface, 'get_trigger_configuration', payload = self.payload_channel)
+
+        if result['n_triggers'] > 1:
+            trigger_type = 'multi_trigger'
+        elif type(result['threshold_high']) == float:
+            trigger_type = 'two_thresholds'
+        elif type(result['threshold']) == float:
+            trigger_type = 'single_threshold'
+        else:
+            trigger_type = None
+
+        return trigger_type
+
+
+    @property
+    def single_threshold_trigger(self):
+        logger.info(self.trigger_type)
+        if self.trigger_type != 'single_threshold':
+            return False
+        threshold = self.provider.cmd(self.psyllid_interface, 'get_trigger_configuration', payload = self.payload_channel)['threshold']
+        return {'threshold' : threshold}
+
+
+    def set_single_threshold_trigger(self, threshold):
```

Comment:
this is a simpler instance of Ben's below comment (requires no changes past the def line):
@single_threshold_trigger.setter
def single_threshold_trigger(self, threshold):